symphony/bdk/core/activity/command.py

```diff
@@ -43,12 +43,41 @@ def bot_display_name(self) -> str:
 
 
 class CommandActivity(AbstractActivity[CommandContext]):
-    """
-    A command activity corresponds to any message sent in a chat where the bot is part of.
+    """A command activity corresponds to any message sent in a chat where the bot is part of.
     """
 
     def matches(self, context: CommandContext) -> bool:
         pass
 
-    def on_activity(self, context: CommandContext):
+    async def on_activity(self, context: CommandContext):
         pass
+
+
+class SlashCommandActivity(CommandActivity):
+    """A slash command is a particular CommandActivity with three parameters:
+      - the command name
+      - if we should mention the bot
+      - the callback function
+
+    The Slash command is triggered if we receive a MESSAGESENT event with message text:
+      - "@{bot_display name} {name}" if ``requires_mention_bot`` is True
+      - "{name}" otherwise
+    """
+
+    def __init__(self, name, requires_mention_bot, callback):
+        """
+
+        :param name: the command name
+        :param requires_mention_bot: if the command requires the bot mention to trigger the slash command
+        :param callback: the coroutine to be executed if message text matches
+        """
+        self._name = name
+        self._requires_mention_bot = requires_mention_bot
+        self._callback = callback
+
+    def matches(self, context: CommandContext) -> bool:
+        pattern = rf"@{context.bot_display_name} {self._name}" if self._requires_mention_bot else rf"{self._name}"
```

Comment:
I was wondering: do we have backslashes in bot names ? If so, I will keep raw strings.

Reply:
The API does not document, I tried to find if there was a check in SBE but I could not find where this is happening
Maybe something to try using ACP?

Reply:
Yes I managed to create such a user: https://devx1.symphony.com/admin-console/#user/12987981108841